core/lib/zksync_core/src/eth_watch/mod.rs

```diff
@@ -55,6 +56,7 @@ impl EthWatch {
         pool: &ConnectionPool,
         poll_interval: Duration,
     ) -> Self {
+        let connection_pool = pool.clone();
```

Comment:
Given that the very first thing you do is clone the pool, probably it makes sense to take an owned object instead of a reference in this method. This way you won't need to clone it inside.

Reply:
Add `drop(storage);` after `Self::initialize_state(..)`.